Lusine_Shahbazyan/web/registration/script.js

```diff
@@ -0,0 +1,131 @@
+class User
+{
+    constructor(uname, passwd, age)
+    {
+        this.username = uname;
+        this.password = passwd;
+        this.age = age;
+    }
+
+    writeComments()
+    {
+        if(this.role == "guests")
+            console.log("I am guest.");
+        else 
+            console.log("I am admin");
+    }
+
+    replyComments()
+    {
+        if(this.role == "guests")
+            console.log("Guests have no permission to reply.");
+        else 
+            console.log("Admin can reply to comments.");
+    }
+}
+
+class Guests extends User
+{
+    role = "guests"
+    constructor(uname, passwd, age, firstname, lastname)
+    {
+        super(uname, passwd, age, firstname, lastname);
+        this.firstname = firstname;
+        this.lastname = lastname;
+    }
+}
+
+class Admin extends User
+{
+    role = "admin";
+    constructor(uname, passwd, age, firstname, lastname)
+    {
+        super(uname, passwd, age, firstname, lastname);
+        this.firstname = firstname;
+        this.lastname = lastname;
+    }
+
+    writeArticles()
+    {
+        console.log("I am admin and i can write articles.")
+    }
+}
+
+function registration(array, obj)
+{
+    if(obj.age < 15)
+    {
+        console.log("Only people over 15 can register");
+        return false;
+    }
+
+    if(array.length === 0)
+    {
+        console.log("Registration completed successfully")
+        return true;
+    }
+
+    for(let i = 0; i < array.length; ++i)
+    {
+        if(obj.username == array[i].username || obj.password == array[i].password)
+        {
+            console.log("There is user with this password and username");
+            return false;
+        }
+    }
+    console.log("Registration completed successfully")
+    return true;
+}
+
+function authentication(array)
```

Comment:
Please try to update authenticate function to have the following form
`authenticate(username, password): return logged in user data`

Reply:
I have update authenticate function